TYPO3.Flow/Classes/TYPO3/Flow/Http/RequestHandler.php

```diff
@@ -129,7 +134,7 @@ public function handleRequest()
      */
     public function getHttpRequest()
     {
-        return $this->request;
+        return $this->componentContext->getHttpRequest();
```

Comment:
Good point, I'll provide a follow-up, though normally you should never call `getHttpRequest()` *before* `handleRequest()` ;)